lib/data/network/media/media_api.dart

```diff
@@ -34,4 +35,20 @@ class UploadFileAPI {
 
     return UploadFileResponse.fromJson(response);
   }
+
+  Future<UrlPreviewResponse> getUrlPreview({
+    required Uri uri,
+    int? preferredPreviewTime,
+  }) async {
+    final response = await _client.get(
+      HomeserverEndpoint.getPreviewUrlServicePath
+          .generateHomeserverMediaEndpoint(),
+      queryParameters: {
+        'url': uri.toString(),
+        if (preferredPreviewTime != null) 'ts': preferredPreviewTime,
+      },
+    ).onError((error, stackTrace) => throw Exception(error));
```

Comment:
don't get this
is it not throw Exception by default ?

Reply:
#652 

this ticket will handle Error details from homeserver

Reply:
i mean, no need onError here